aptos-move/framework/aptos-framework/sources/aggregator_v2/aggregator_v2.move

```diff
@@ -0,0 +1,48 @@
+/// This module provides an interface for aggregators (version 2).
+/// Only skeleton - for AggregagtorSnapshot - is provided at this time,
+/// to allow transition of usages.
+module aptos_framework::aggregator_v2 {
+    use std::string::String;
+
+    struct AggregatorSnapshot<Element> has store, drop {
+        value: Element,
+    }
+
+    public native fun create_snapshot<Element: copy + drop>(value: Element): AggregatorSnapshot<Element>;
+
+    public native fun copy_snapshot<Element: copy + drop>(snapshot: &AggregatorSnapshot<Element>): AggregatorSnapshot<Element>;
+
+    public native fun read_snapshot<Element>(snapshot: &AggregatorSnapshot<Element>): Element;
+
+    public native fun string_concat<Element>(before: String, snapshot: &AggregatorSnapshot<Element>, after: String): AggregatorSnapshot<String>;
+
+    #[test]
+    public fun test_correct_read() {
+        let snapshot = create_snapshot(42);
+        let snapshot2 = copy_snapshot(&snapshot);
+        assert!(read_snapshot(&snapshot) == 42, 0);
+        assert!(read_snapshot(&snapshot2) == 42, 0);
+    }
+
+    #[test]
+    public fun test_correct_read_string() {
+        let snapshot = create_snapshot(std::string::utf8(b"42"));
+        let snapshot2 = copy_snapshot(&snapshot);
+        assert!(read_snapshot(&snapshot) == std::string::utf8(b"42"), 0);
+        assert!(read_snapshot(&snapshot2) == std::string::utf8(b"42"), 0);
+    }
+
+    #[test]
+    #[expected_failure]
+    public fun test_incorrect_string_concat() {
+        let snapshot = create_snapshot(42);
+        string_concat(std::string::utf8(b"before"), &snapshot, std::string::utf8(b"after"));
```

Comment:
asserting on error code would be nice :) I think you can add it to `expected_failure` above

Reply:
I might be missing something, but why does this throw an exception? because of a wrong type?

we don't have any way to convert AggeragorSnapshot<u64> to AggregatorSnapshot<Sting>, so this would make it useless, and we wouldn't be able to create a string that contains an index?

if we think converting in string_concat is confusing, let's just change to format_one_snapshot() that takes any snapshot type and puts it into a string